core/flamingock-test-support/src/main/java/io/flamingock/internal/core/builder/WhenStage.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2025 Flamingock (https://www.flamingock.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.flamingock.internal.core.builder;
+
+import java.util.function.Consumer;
+import java.util.List;
+import io.flamingock.internal.common.core.audit.AuditEntry;
+
+public interface WhenStage {
+    ThenStage thenExpectAuditSequenceStrict(AuditEntryExpectation... expectations);
+    ThenStage thenExpectException(Class<? extends Throwable> exceptionClass, Consumer<Throwable> validator);
+    ThenStage thenInspectAuditEntries(Consumer<List<AuditEntry>> validator);
+
+    void verify();
+}
```

Comment:
This interface( and the implementation class) is not needed.


with the current API we can verify nothing, because it allows
```java 
FlamingockTestSupport
  .given(builder)
//...
  .whenRun()
  .verify();
```

So first thing would be removing the method `verify` from here.

But this interface only makes sense if we want to have more than one **when action**, however by definition there should only be one action. So this interface gives nothing really.

My suggestion is removing this interface(and the implementation class) and the `whenRun` method in the `GivenStage` interface returns directly a `ThenStage`